lib/ansible/modules/cloud/google/gcp_target_proxy.py

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/python
+# Copyright 2017 Google Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: gcp_target_proxy
+version_added: "2.4"
+short_description: Create, Update or Destory a Target_Proxy.
+description:
+    - Create, Update or Destory a Target_Proxy. See
+      U(https://cloud.google.com/compute/docs/load-balancing/http/target-proxies) for an overview.
+      More details on the Target_Proxy API can be found at
+      U(https://cloud.google.com/compute/docs/reference/latest/targetHttpProxies#resource-representations).
+requirements:
+  - "python >= 2.6"
+  - "google-api-python-client >= 1.6.2"
+  - "google-auth >= 0.9.0"
+  - "google-auth-httplib2 >= 0.0.2"
+notes:
+  - Currently only supports global HTTP proxy.
+author:
+  - "Tom Melendez (@supertom) <tom@supertom.com>"
+options:
+  target_proxy_name:
+    description:
+       - Name of the Target_Proxy.
+    required: true
+  target_proxy_type:
+    description:
+       - Type of Target_Proxy. HTTP, HTTPS or SSL. Only HTTP is currently supported.
+    required: true
+  url_map:
```

Comment:
There isn't anything else it could be referred to as.  Still, to be consistent I renamed it.